### PR TITLE
[REACTOS] *.spec: Fix some function/parameter types

### PR DESCRIPTION
**Pull request by SergeGautherie**

### CODE REVIEW

**Review by learn-more — APPROVED** [review]

File: dll/win32/advapi32/advapi32.spec [hunk 1]
```diff
@@ -631,8 +631,8 @@
 @ stdcall SystemFunction041(ptr long long) # RtlDecryptMemory
 @ stdcall TraceEvent(double ptr) ntdll.EtwTraceEvent
 @ stdcall TraceEventInstance(double ptr ptr ptr) ntdll.EtwTraceEventInstance
-@ varargs TraceMessage(ptr long ptr long) ntdll.EtwTraceMessage
-@ stdcall TraceMessageVa(double long ptr long ptr) ntdll.EtwTraceMessageVa
+@ varargs TraceMessage() ntdll.EtwTraceMessage
+@ stdcall TraceMessageVa() ntdll.EtwTraceMessageVa
```
Comment (HBelusca) [inline dll/win32/advapi32/advapi32.spec:635]: Please keep the parameters inside for consistency?

File: win32ss/user/user32/user32.spec [hunk 2]
```diff
@@ -744,7 +744,7 @@
 735 stdcall WindowFromPoint(double)
 736 stdcall keybd_event(long long long long)
 737 stdcall mouse_event(long long long long long)
-738 varargs wsprintfA(str str)
-739 varargs wsprintfW(wstr wstr)
+738 varargs wsprintfA(ptr str)
+739 varargs wsprintfW(ptr wstr)
```
Comment (HBelusca) [inline win32ss/user/user32/user32.spec:748]: Why is the first parameter "ptr"? It's a pointer to string, much like the second parameter....

File: dll/win32/shell32/shell32.spec [hunk 3]
```diff
@@ -178,8 +178,8 @@
 179 stdcall SHGetNewLinkInfoA(str str ptr long long)
 180 stdcall SHGetNewLinkInfoW(wstr wstr ptr long long)
 181 stdcall -noname RegisterShellHook(long long)
-182 varargs ShellMessageBoxW(long long wstr wstr long)
-183 varargs ShellMessageBoxA(long long str str long)
+182 varargs ShellMessageBoxW(ptr ptr wstr wstr long)
```
Comment (HBelusca) [inline dll/win32/shell32/shell32.spec:181]: I think there will be a conflict here wrt. master.

**Review by HBelusca — CHANGES_REQUESTED** [review]

File: win32ss/user/user32/user32.spec (hunk 2 above)
Comment (ThFabba) [reply win32ss/user/user32/user32.spec:748]: It's a pointer to a buffer, i.e. it's not initialized on input. Remember that the `ptr` vs `str` distinction is for relay tracing.

Comment (HBelusca) [reply win32ss/user/user32/user32.spec:748]: Thanks for the info!

File: dll/win32/advapi32/advapi32.spec (hunk 1 above)
Comment (SergeGautherie) [reply dll/win32/advapi32/advapi32.spec:635]: I too used to assume that, then I was told not to, as it is useless for forwarders: less code to maintain...

File: dll/win32/shell32/shell32.spec (hunk 3 above)
Comment (SergeGautherie) [reply dll/win32/shell32/shell32.spec:181]: Yeah, rebased.

**Pull request merged**